subsys/nrf_security/src/drivers/cracen/cracenpsa/cracenpsa.cmake

```diff
@@ -63,6 +63,19 @@ if(CONFIG_PSA_NEED_CRACEN_MAC_DRIVER)
   list(APPEND cracen_driver_sources
     ${CMAKE_CURRENT_LIST_DIR}/src/mac.c
   )
+
+  if(CONFIG_PSA_NEED_CRACEN_HMAC)
```

Comment:
Not an expert on the build system so might not be needed, but should these includes not depend on the `CONFIG_PSA_CRACEN_MAC_DRIVER ` as well? As at least ECDSA uses the `CONFIG_PSA_NEED_CRACEN_HMAC` without needing `cracen_mac_hmac.c`

Reply:
I am afraid I cannot follow this, sorry.  ECDSA includes the hmac.c file without being affected by the PSA_NEEDs for MAC. So I am not sure what you mean here.

Reply:
Ah. I just mean that there are cases where `CONFIG_PSA_NEED_CRACEN_HMAC` is true, but you don't need `cracen_mac_hmac.c` which is appended here.  `cracen_mac_hmac.c` is only needed when using `mac` and `hmac.` So it is more permissive than it needs to be. I don't think it makes much of a practical difference. Hopefully that is clearer.

Thinking a bit more about it the correct choice would probably be to remove `hmac.c` from being added with `CONFIG_PSA_NEED_CRACEN_ASYMMETRIC_SIGNATURE_DRIVER` and having just `hmac.c` under `CONFIG_PSA_NEED_CRACEN_HMAC` and then having `cracen_mac_hmac.c` under `if(CONFIG_PSA_NEED_CRACEN_MAC_DRIVER AND CONFIG_PSA_NEED_CRACEN_HMAC` but that is outside the scope of this pr for sure.